File: Sharebook/Server/src/Services/Post/Sharebook.Post.Domain/AggregatesModel/PostAggregate/Entities/IUserEntities.cs
﻿namespace Sharebook.Post.Domain.AggregatesModel.PostAggregate.Entities
{
    public interface IUserEntities
    {
        /// <summary>
        /// Website metadata
        /// </summary>
        IWebsiteEntity Website { get; set; }

        /// <summary>
        /// User description
        /// </summary>
        IDescriptionEntity Description { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Entities/IUserEntities.ts
import {InjectionToken} from "@angular/core";

import {IDescriptionEntity} from "./IDescriptionEntity";
import {IWebsiteEntity} from "./IWebsiteEntity";
import {UserEntities} from "../../../Core/Models/TwitterEntities/UserEntities";

export interface IUserEntities {
  // Website metadata
  website: IWebsiteEntity;

  // User description
  description: IDescriptionEntity;
}

export const IUserEntitiesToken = new InjectionToken<IUserEntities>('IUserEntities', {
  providedIn: 'root',
  factory: () => new UserEntities(),
});
